rfcs/0035-create-connection.md

```diff
@@ -0,0 +1,143 @@
+---
+feature: Create Connection
+authors:
+  - "Bohan Zhang"
+start_date: "2023/01/15"
+---
+
+# RFC: Create Connection
+
+## Motivation
+
+A connection is used to describe how to connect to an external system that users want to read data from. Once created, a connection is reusable across `CREATE SOURCE` and `CREATE SINK` statements. This RFC proposes a new `CREATE CONNECTION` statement to create a connection.
+
+## Design
+
+### Syntax
+
+```sql
+-- create connection
+CREATE CONNECTION (IF NOT EXIST) {{ connection_name }}
+  TO {{ connection_type }}
+  ( {{ field }} = {{ value }}, ...);
+
+-- create source with connection
+CREATE SOURCE {{ source_name }} ( {{ field }} = {{ value }}, ... )
+  FROM CONNECTION {{ connection_name }}
```

Comment:
’With connection’ may be better than ‘from/to connection’ because: 

1. It’s consistent between source & sink
2. To avoid the potential ambiguity of the word ‘from’